burn-core/src/optim/adam.rs

```diff
@@ -88,7 +93,12 @@ impl AdamConfig {
             },
             weight_decay: self.weight_decay.as_ref().map(WeightDecay::new),
         };
-        OptimizerAdaptor::from(optim)
+
+        let mut optim = OptimizerAdaptor::from(optim);
+        if let Some(config) = &self.grad_clipping {
```

Comment:
@nathanielsimard Can you explain how config works? I tried playing with it previously but only got it to work with a weird struct setup. Is all that's required that the config have a default?

Reply:
Not really, config can be enum (no default yet) or struct. Each member of the config needs to be (de)serializable, that's it. It's normaly a good idea to separate the config type from the actual type. In my case the type are pretty similar (two enums) but they may evolve differently and you don't have to handle a None case.